Move compute mirror logic into async.

dbatiste opened the pull request:
@dlockhart : this change wraps the compute mirror logic in `this.async` which will ensure that child components (in this case `iron-icon`) are ready when this logic runs.  There is no guarantee that the order components are `ready` in a particular order.

Background: I ran into timing issue where the `svg` was not distributed yet when this logic was run.  Wrapping in this component's `async` resolves the issue.

dbatiste commented:
Following our conversation, I am going to update to handle changes to the icon as well, which affects this change a little bit... so more change coming.

dlockhart commented on d2l-icon.html:
```diff
@@ -33,10 +33,12 @@
+					if (mirrorRtl) {
+						this.$$('iron-icon').setAttribute('data-mirror-rtl', '');
+					}
+				});
```
A little surprised this works without binding the async callback to `this`...

dbatiste commented:
I've updated things a bit here:
* added support for changing `icon` attr on `d2l-icon`
* added some tests (they know a bit about internals to enable testing, but i think worthwhile)
* updated demo page

dbatiste commented on d2l-icon.html:
```diff
@@ -64,7 +63,32 @@
+				if (mirrorRtl) {
+					this.$$('iron-icon').setAttribute('data-mirror-rtl', '');
+				} else {
+					this.$$('iron-icon').removeAttribute('data-mirror-rtl');
```
It was necessary to add this `removeAttribute` call because the same `iron-icon` instance is used.

dlockhart commented on d2l-icon.html:
```diff
@@ -64,7 +63,32 @@
+				svg.setAttribute('focusable', 'false');
+				var mirrorRtl = this.__computeMirrorRtl(svg);
+				if (mirrorRtl) {
+					this.$$('iron-icon').setAttribute('data-mirror-rtl', '');
```
Should `__getIronIcon()` be called here or can it just be removed?

dbatiste replied:
Ah good eye!  So, I created this helper just to facilitate the unit tests.

dbatiste replied:
Ideally I would not need such a helper.  A perceptual diff would enable testing the flip case better.  :)

dlockhart commented:
Looks great @dbatiste, thanks for fixing this up and adding the tests! Next: Galen perceptual diffs...
